qiskit_experiments/calibration/calibrations.py

```diff
@@ -0,0 +1,691 @@
+
+        return circ
+
+    def schedules(self) -> List[Dict[str, Any]]:
+        """
+        Return the schedules in self in a list of dictionaries to help
+        users manage their schedules.
+
+        Returns:
+            data: A list of dictionaries with all the schedules in it.
+        """
+        data = []
+        for context, sched in self._schedules.items():
+            data.append({"context": context, "schedule": sched, "parameters": sched.parameters})
+
+        return data
+
+    def parameters_table(
+        self,
+        parameters: List[str] = None,
+        schedules: List[Union[Schedule, str]] = None,
+        qubit_list: List[Tuple[int, ...]] = None,
+    ) -> List[Dict[str, Any]]:
+        """
+
+        Args:
+            parameters: The parameter names that should be included in the returned
+                table. If None is given then all names are included.
+            schedules: The schedules to which to restrict the output.
+            qubit_list: The qubits that should be included in the returned table.
+                If None is given then all channels are returned.
+
+        Returns:
+            data: A dictionary of parameter values which can easily be converted to a
+                data frame.
+        """
+
+        data = []
+
+        # Convert inputs to lists of strings
+        if parameters is not None:
+            parameters = {prm.name if isinstance(prm, Parameter) else prm for prm in parameters}
+
+        if schedules is not None:
+            schedules = {sdl.name if isinstance(sdl, Schedule) else sdl for sdl in schedules}
+
+        # Look for exact matches. Default values will be ignored.
+        keys = set()
+        for key in self._params.keys():
+            if parameters and key.parameter not in parameters:
+                continue
+            if schedules and key.schedule not in schedules:
+                continue
+            if qubit_list and key.qubits not in qubit_list:
+                continue
+
+            keys.add(key)
+
+        for key in keys:
+            for value in self._params[key]:
+                value_dict = dataclasses.asdict(value)
+                value_dict["qubits"] = key.qubits
+                value_dict["parameter"] = key.parameter
+                value_dict["schedule"] = key.schedule
+
+                data.append(value_dict)
+
+        return data
+
+    def to_csv(self):
+        """
+        Serializes the parameterized schedules and parameter values so
+        that they can be stored in csv file.
+        """
+
+        # Write the parameter configuration.
+        header_keys = ["parameter.name", "hash(parameter)", "schedule", "qubits"]
+        body = []
+
+        for parameter, keys in self.parameters.items():
+            for key in keys:
+                body.append(
+                    {
+                        "parameter.name": parameter[0].name,
+                        "hash(parameter)": parameter[1],
```

Comment:
I'm still not fully convinced with the usage of hash value. This hash value can change after you reload the database, due to random uuid generation in the parameter object constructor. I think this is very confusing for end-users. If we stick to the parameter object, we should extract the uuid value rather than hash, then save it and override the default uuid with the stored value to create the completely same object.

Reply:
To me, the benefit of including the hash here is that it allows you to identify parameters that were linked. So `(amp, xp, (0,)` and `(amp, xm, (0,))` would have the same hash if `xm` were defined with the same `Parameter`. This role would be replaced by a serialization method for the schedules.

Reply:
I think we can include a `_hash_map` variable that `_register_parameter` will manage.

Reply:
This should improve the situation: f817f5c